readthedocs/doc_builder/director.py

```diff
@@ -225,7 +225,7 @@ def checkout(self):
         log.info("Cloning and fetching.")
         self.vcs_repository.update()
 
-        identifier = self.data.build_commit or self.data.version.identifier
+        identifier = self.data.build_commit or self.data.version.commit_name
```

Comment:
It would be good to put a comment here explaining this line. Mainly why `commit_name` is the correct one instead of `identifier`.

Besides, we should double check that this logic does not interferes with what we have written in https://github.com/readthedocs/readthedocs.org/blob/996a5b2094ec6fdcfa8a167588bb69137f20056b/readthedocs/vcs_support/backends/git.py#L91-L98

that has a similar logic to decide what's the correct "thing" to fetch/clone.

Reply:
yeah, both are kind of similar, but just different enough that I wasn't able to merge them.

Reply:
Great comment! 💯